tests/test_operators.py

```diff
@@ -26,6 +34,56 @@ def test_fractional_power_unitary_hermitian_operators(unitary_hermitian_operator
         assert sqrt(operator(0)) * sqrt(operator(1)) == sqrt(operator(0)) * sqrt(operator(1))
 
 
-def test_parametric_opertor() -> None:
+def test_parametric_operator() -> None:
     theta = variable("theta")
     assert RX(theta / 2)() * RX(theta / 2)() == RX(theta)()
+
+
+## Analog Operators
+
+
+def test_native_drive() -> None:
+    drive = NativeDrive(1.0, 1.0, 1.0, 0.0)
+    assert drive().is_quantum_operator
+
+
+def test_piecewise_drive() -> None:
+    drive = PiecewiseDrive(
+        array_parameter("duration", size=2),
+        array_parameter("amplitude", size=3),
+        array_parameter("detuning", size=3),
+    )
+    assert drive().is_quantum_operator
```

Comment:
I guess it would be a bit neater to set up different cases for the different failure modes and then assert if the correct exception is raised.